File: app/src/main/java/com/baronzhang/android/weather/base/BaseRecyclerViewAdapter.java
package com.baronzhang.android.weather.base;

import android.support.v7.widget.RecyclerView;
import android.widget.AdapterView;

/**
 * @author baronzhang (baron[dot]zhanglei[at]gmail[dot]com)
 *         16/4/15
 */
public abstract class BaseRecyclerViewAdapter<T extends RecyclerView.ViewHolder> extends RecyclerView.Adapter<T>{

    protected AdapterView.OnItemClickListener onItemClickListener;

    public void setOnItemClickListener(AdapterView.OnItemClickListener onItemClickListener) {

        this.onItemClickListener = onItemClickListener;
    }

    protected void onItemHolderClick(RecyclerView.ViewHolder itemHolder) {
        if (onItemClickListener != null) {
            onItemClickListener.onItemClick(null, itemHolder.itemView,
                    itemHolder.getAdapterPosition(), itemHolder.getItemId());
        } else {
            throw new IllegalStateException("Please call setOnItemClickListener method set the click event listeners");
        }
    }

}

File: app/src/main/java/com/baronzhang/android/weather/base/BaseRecyclerViewAdapter.kt
package com.baronzhang.android.weather.base

import androidx.recyclerview.widget.RecyclerView
import android.widget.AdapterView

/**
 * @author baronzhang (微信公众号：BaronTalk)
 */
abstract class BaseRecyclerViewAdapter<T : RecyclerView.ViewHolder> : RecyclerView.Adapter<T>() {

    var onItemClickListener: AdapterView.OnItemClickListener? = null

    protected fun onItemHolderClick(itemHolder: RecyclerView.ViewHolder) {
        if (onItemClickListener != null) {
            onItemClickListener!!.onItemClick(null, itemHolder.itemView,
                    itemHolder.adapterPosition, itemHolder.itemId)
        } else {
            throw IllegalStateException("Please call setOnItemClickListener method set the click event listeners")
        }
    }

}
